src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/serviceprovider/ReferralReportJobListener.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.serviceprovider
+
+import mu.KLogging
+import net.logstash.logback.argument.StructuredArguments
+import org.springframework.batch.core.BatchStatus
+import org.springframework.batch.core.JobExecution
+import org.springframework.batch.core.JobExecutionListener
+import org.springframework.stereotype.Component
+import java.io.File
+import kotlin.io.path.createTempDirectory
+
+@Component
+class ReferralReportJobListener() : JobExecutionListener {
+  companion object : KLogging()
+
+  override fun beforeJob(jobExecution: JobExecution) {
+    // create a temp file for the job to write to and store the path in the execution context
+    val params = jobExecution.jobParameters.parameters
+    val path = createTempDirectory().resolve("${params["user.id"]}_${params["timestamp"]}.csv")
```

Comment:
Just to check: Are you not passing in `from` and `to` params instead of `timestamp`?

Reply:
we pass to and from as well, in order to build the query, but the timestamp and the requesting user uniquely identifies the job, and therefore the output file too.